orderbook/src/api/filter.rs

```diff
@@ -112,8 +113,16 @@ pub mod test_util {
         };
         let response = post().await;
         assert_eq!(response.status(), StatusCode::CREATED);
+        let body: handler::UidResponse = serde_json::from_slice(response.body()).unwrap();
+
+        assert_eq!(body, serde_json::from_value(expected_uid).unwrap());
         // Posting again should fail because order already exists.
         let response = post().await;
         assert_eq!(response.status(), StatusCode::BAD_REQUEST);
+
+        let body: handler::OrderPostError = serde_json::from_slice(response.body()).unwrap();
```

Comment:
I think the type of body should be `serde_json::Value` so that it can be directly compared to `expected_error` without first converting `expected_error` to OrderPostError.